forwarder/kafka/kafka_producer.py

```diff
@@ -32,6 +34,11 @@ def produce(
         def ack(err, _):
             if err:
                 self.logger.error(f"Message failed delivery: {err}")
+            else:
+                # increment only for PVs related updates
+                # key is None when we send commands?
+                if self._update_msg_counter and key is not None:
+                    self._update_msg_counter.increment()
 
         self._producer.produce(
```

Comment:
Since `produce` method of `KafkaProducer` also has `key` parameter, How about we use Queue solution and use `pv_name` here to put `f"{pv_name}"` in the queue. In StatisticReporter we clear the Queue and counts numbers for each `pv_name`. Therefore we can send total number of updates for each PV to graphite server.

Reply:
> total number of updates for each PV

Unless someone has specifically asked for this level of detail, I think it would be better to keep it very simple and only track total number of outgoing messages. That makes the implementation much easier because no queue or registering of counters with the reporter is needed.